### PR TITLE
[Minor] List of agent's enabled modules added to heartbeat according to issue #229

### PR DESCRIPTION
**Pull request by Caiyeon**
Adds functionality as per issue #229

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ type Agent struct {
 	Authorized      bool        `json:"authorized,omitempty"`
 	Env             AgentEnv    `json:"environment,omitempty"`
 	Tags            interface{} `json:"tags,omitempty"`
+	Modules         []string    `json:"modules,omitempty"`
```
Comment (ameihm0912) [inline agent.go]: I'm wondering if we should include this information in the agent's environment element here (Env), vs. putting it in a new part of the heartbeat. The advantage of putting it in the environment is we get this information inserted into the database without any changes there. If we want to put it outside the environment, I think we should modify the scheduler code to also insert this information into the agents table.

See for example here: https://github.com/mozilla/mig/blob/master/mig-scheduler/agents_queues.go#L125 where we process the incoming heartbeat, and here: https://github.com/mozilla/mig/blob/master/database/agents.go#L97

One of the goals with this is to be able to target agents running specific modules via -t to the command line and so on, so we will need this information in the database.

**Comment by Caiyeon** [comment]
I've relocated the module updates to mig-agent/context.go; This seems to be closer to the ideal location. I have a couple of concerns.

Is it possible that a module starts up with the agent, but fails later on?

At the moment everything in the AgentEnv comes from AgentContext. AgentContext's principal is to only gather information regarding the system, rather than the agent itself. Thus, by placing the module update in this location, the "env has changed" return value is questioned. If the module availability changes, has the environment changed? AgentContext has not changed, but AgentEnv has changed.

**Comment by ameihm0912** [comment]
> Is it possible that a module starts up with the agent, but fails later on?

It's possible, but the agent will keep trying to restart it. Right now, we don't have any sort of concept of a "disabled" module state. So, modules.Available will be static and shouldn't change given a build of the agent.

I see what you mean about AgentContext. As you mentioned this is really meant for grabbing information about the system that can change. Some of the data in ctx.Agent.Env can change based on the AgentContext, but some of it stays the same when the agent runs (like Env.Proxies).

I'm wondering if we can treat this like we do the Proxy related values, and just set it on startup? For example, this is where the proxy values get set, https://github.com/mozilla/mig/blob/master/mig-agent/context.go#L493-L494.

Could we just set env.Modules somewhere in Init() for example?

**Comment by Caiyeon** [comment]
Placed it in Init(). I originally placed it in the update function thinking it wasn't static.

While this allows targeting agents with specific modules, it does not guarantee that the agent's specified module is fully functional. But I suppose with the current structure this is as close as it gets.

**Comment by ameihm0912** [comment]
@Caiyeon thanks, this looks good to go

Can we just amend the commit according to https://github.com/mozilla/mig/blob/master/CONTRIBUTING.md?

We add a tag to the front of the commit message indicating the scope of the change, so for example in this case I'd suggest [minor]. Once we have that I think it's ready to merge.

**Pull request merged**